app/javascript/packages/document-capture/components/form-steps.jsx

```diff
@@ -211,8 +174,8 @@ function FormSteps({ steps = [], onComplete = () => {}, initialValues = {}, init
     headingRef.current?.focus();
   }
 
-  const { form: Form, footer: Footer, name, title } = effectiveStep;
-  const isLastStep = effectiveStepIndex + 1 === steps.length;
+  const { form: Form, footer: Footer, name, title } = step;
+  const isLastStep = stepIndex + 1 === steps.length;
```

Comment:
Random thought... The steps are fixed per load, right? We show/hide steps based on desktop/mobile, but we don't change the steps *during* the flow right?

Reply:
> Random thought... The steps are fixed per load, right? We show/hide steps based on desktop/mobile, but we don't change the steps _during_ the flow right?

Correct, with the one additional caveat that when a submission fails, we'll re-mount `FormSteps`. Currently, we remount with the same steps, but as of #4237 this will be a separate set of steps (an array of the one "Review" step).

Was there something about this specific code that caught your attention relevant to the steps being fixed?

Reply:
> Was there something about this specific code that caught your attention relevant to the steps being fixed?

No, I didn't remember if we always detected the end like this, so it got me pondering about if the end condition would trigger on accident or something